packages/pluggableWidgets/line-chart-native/src/components/LineChart.tsx

```diff
@@ -0,0 +1,163 @@
+import { View, LayoutChangeEvent, Text } from "react-native";
+import { InterpolationPropType } from "victory-core";
+import { VictoryChart, VictoryLine, VictoryGroup, VictoryScatter, VictoryAxis, VictoryLabel } from "victory-native";
+
+import { LineChartStyle } from "../ui/Styles";
+import { Legend } from "./Legend";
+
+export interface LineChartProps {
+    series: LineChartSeries[];
+    style: LineChartStyle;
+    showLegend: boolean;
+    xAxisLabel?: string;
+    yAxisLabel?: string;
+}
+
+export interface LineChartSeries {
+    dataPoints: LineChartDataPoints;
+    interpolation: InterpolationPropType;
+    name?: string;
+    lineStyle: "line" | "lineWithMarkers" | "custom";
+    stylePropertyName?: string;
+}
+
+export type LineChartDataPoints =
+    | Array<LineChartDataPoint<number, number>>
+    | Array<LineChartDataPoint<number, Date>>
+    | Array<LineChartDataPoint<Date, number>>
+    | Array<LineChartDataPoint<Date, Date>>;
+
+export interface LineChartDataPoint<X extends number | Date, Y extends number | Date> {
+    x: X;
+    y: Y;
+}
+
+export function LineChart(props: LineChartProps): ReactElement | null {
+    const { series, showLegend, style, xAxisLabel, yAxisLabel } = props;
+
+    if (series.length === 0) {
+        return null;
+    }
+
+    const dataTypesResult = useMemo(() => getDataTypes(series), [series]);
+
+    const chartLines = useMemo(() => {
+        if (!dataTypesResult || dataTypesResult instanceof Error) {
+            return null;
+        }
+
+        return series.map((series, index) => {
+            const { dataPoints, interpolation, lineStyle, stylePropertyName } = series;
+
+            const seriesStyle = style.series && stylePropertyName ? style.series[stylePropertyName] : undefined;
+
+            const markers =
+                lineStyle === "lineWithMarkers" ||
+                (lineStyle === "custom" && seriesStyle?.markers?.display !== "false") ? (
+                    <VictoryScatter data={dataPoints} style={seriesStyle?.markers} size={seriesStyle?.markers?.size} />
+                ) : (
+                    undefined
+                );
+
+            return (
+                <VictoryGroup key={index}>
+                    {markers && (!seriesStyle?.markers?.display || seriesStyle.markers.display !== "onTop")
+                        ? markers
+                        : null}
+                    <VictoryLine style={seriesStyle?.line} data={dataPoints} interpolation={interpolation} />
+                    {markers && seriesStyle?.markers?.display === "onTop" ? markers : null}
+                </VictoryGroup>
+            );
+        });
+    }, [dataTypesResult, series, style]);
+
+    const [chartDimensions, setChartDimensions] = useState<{ height: number; width: number }>();
+
+    const updateChartDimensions = useCallback((event: LayoutChangeEvent) => {
+        const { height, width } = event.nativeEvent.layout;
+        setChartDimensions({ height, width });
+    }, []);
```

Comment:
I am pretty sure you need to add setChartDimensions as dependency otherwise it could cause re-renders with old data

Reply:
Changed it. Also in src/LineChart.tsx line 20